Content.Server/GameObjects/Components/Sound/BaseEmitSoundComponent.cs

```diff
@@ -0,0 +1,27 @@
+using Robust.Shared.GameObjects;
+using Robust.Shared.Serialization.Manager.Attributes;
+using Robust.Shared.ViewVariables;
+
+namespace Content.Server.GameObjects.Components.Sound
+{
+    public enum EmitSoundMode
+    {
+        Auto,
+        Single,
+        RandomFromCollection
+    }
+
+    /// <summary>
+    /// Base sound emitter which defines most of the data fields.
+    /// Default EmitSoundMode is Auto, which will first try to play a sound collection,
+    /// and if one isn't assigned, then it will try to play a single sound.
+    /// </summary>
+    public abstract class BaseEmitSoundComponent : Component
+    {
+        [ViewVariables(VVAccess.ReadWrite)] [DataField("sound")] public string? _soundName;
```

Comment:
give these datafieldsfields default values, that should allow you to remove the `pragma warning disable CS8604` in EmitSoundSystem

Reply:
used !string.IsNullOrEmpty() + logging, I think it will be better for debugging later, since we will not be masking null values with default values. But I won't insist on this approach, if you guys think default values are better